src/benchmark_runner.cc

```diff
@@ -203,6 +207,17 @@ class BenchmarkRunner {
     std::unique_ptr<internal::ThreadManager> manager;
     manager.reset(new internal::ThreadManager(b.threads));
 
+#ifdef _OPENMP
+    if (b.threads == OmpNoParallelRegion) {
+      RunInThread(&b, iters, 0, manager.get());
+    }
+    else {
+    #pragma omp parallel num_threads(b.threads)
+      {
+        RunInThread(&b, iters, omp_get_thread_num(), manager.get());
+      }
+    }
```

Comment:
I would think this should be closer to
```c++
// Start specified thread count, so that the benchmark, if it uses threads internally, doesn't have to.
#pragma omp parallel num_threads(b.thread_count)
{
  // Do we actually want to run the same thing in each thread?
  if(!(b.threading & OmpNoParallelRegion)) {
    // Alright then, in each thread, run the benchmark.
    RunInThread(&b, iters, omp_get_thread_num(), manager.get());
  } else {
    // If not, then just do it it a single thread.
#pragma omp single
    RunInThread(&b, iters, omp_get_thread_num(), manager.get());
  }
}
```

Reply:
Before I finish my changes, just let us define possibple scenarios:
1. No threading.
2. Use std::thread (as usual).
3. Open an openmp region. Threads(...) will determine the number of threads in that region.
4. Don't open an openmp region, but just set the number of threads via omp_set_num_threads(). The benchmark will open the region by itself.

Do we agree, that the list is complete regarding this PR? 
Do we agree, that the framework does not support a combination of mode 2 with any of the other modes - e.g. no one wants n std::threads with m openmp threads?

How do we expose these modes to the user? I see three options:
1. We encode it in the Threads argument(s) via some bit flags. E.g. `->Threads(OmpRegion|8)` or `->ThreadRange(OmpRegion|4, OmpRegion|8)`.
2. We have dedicated functions like `->UseOpenMP()->`ThreadRange(4, 8)`
3. The mode is a (defaulted) argument to the Threads function family. E.g.  `->Threads(8, OmpRegion)` or `->ThreadRange(4, 8, OmpRegion)`.

Out of these three options I like the third one most. Do you see another approach?

Reply:
Will reply to the rest later, but 

> * We have dedicated functions like `->UseOpenMP()->ThreadRange(4, 8)`

^ this

Reply:
At a first glance this seems to be the best option. However note, that this would fix the whole benchmark family to one mode. Thus, if you like to test another threading mode too, you have to re-register the benchmark with all other arguments being the same. 

How about a fourth option: use dedicated functions, e.g. `->OmpThreads(8)` or `->OmpThreadRange(4, 8)`?
This approach isn't extensible that nicely to other parallelization APIs though.

Reply:
> At a first glance this seems to be the best option. However note, that this would fix the whole benchmark family to one mode. Thus, if you like to test another threading mode too, you have to re-register the benchmark with all other arguments being the same.

Yep.

> Or do you want to have an ordering semantic? E.g.
> 
> ```
> BM(...)->Threads(2)->UseOpenMP()->Threads(8);
> ```
> 
> tests BM with 2 std::threads and then with 8 OpenMP threads. I don't like this.

No, i don't want not like this.